src/tests/ftest/container/Attribute.py

```diff
@@ -100,6 +101,7 @@ def setUp(self):
         self.hostlist = self.params.get("test_machines", '/run/hosts/*')
         self.hostfile = WriteHostFile.WriteHostFile(self.hostlist, self.workdir)
 
+        self.agent_session = AgentUtils.run_agent(basepath, self.hostlist)
```

Comment:
You also need to initialise `self.agent_session = None` at the start of `setUp()` because it is used in `tearDown()`.  One of the `sed` expressions from my last review does that.  You could still use just that expression to easily add the init to all of the required `setUp()`s.